Signal/src/Loki/View Controllers/DeviceLinkingModal.swift

```diff
@@ -184,13 +184,16 @@ final class DeviceLinkingModal : Modal, DeviceLinkingSessionDelegate {
         let signedDeviceLink = DeviceLink(between: master, and: deviceLink.slave)
         LokiFileServerAPI.addDeviceLink(signedDeviceLink).done(on: DispatchQueue.main) { [weak self] in
             SSKEnvironment.shared.messageSender.send(linkingAuthorizationMessage, success: {
-                let thread = TSContactThread.getOrCreateThread(contactId: deviceLink.slave.hexEncodedPublicKey)
+                let hexEncodedPublicKey = deviceLink.slave.hexEncodedPublicKey
+                let thread = TSContactThread.getOrCreateThread(contactId: hexEncodedPublicKey)
```

Comment:
This line potentially starts a new write transaction, and the line below definitely does. It'd be nice to make that explicit and also combine them together for a slight performance benefit.